client/app/dbaas/logs/detail/aliases/logs-aliases.service.js

```diff
@@ -0,0 +1,223 @@
+     *
+     * @param {any} serviceName
+     * @param {any} aliasId
+     * @returns promise which will be resolve to alias object
+     * @memberof LogsAliasesService
+     */
+    getAapiAlias (serviceName, aliasId) {
+        return this.AliasAapiService.get({ serviceName, aliasId })
+            .$promise.catch(this.ServiceHelper.errorHandler("logs_alias_get_error"));
+    }
+
+    /**
+     * returns objecy containing total number of aliases and total number of aliases used
+     *
+     * @param {any} serviceName
+     * @returns quota object containing V (total number aliases) and configured (number of aliases used)
+     * @memberof LogsAliasesService
+     */
+    getQuota (serviceName) {
+        return this.AccountingAapiService.me({ serviceName }).$promise
+            .then(me => {
+                const quota = {
+                    max: me.total.maxNbAlias,
+                    configured: me.total.curNbAlias,
+                    currentUsage: me.total.curNbAlias * 100 / me.total.maxNbAlias
+                };
+                return quota;
+            }).catch(this.ServiceHelper.errorHandler("logs_alias_quota_get_error"));
+    }
+
+    /**
+     * delete alias
+     *
+     * @param {any} serviceName
+     * @param {any} alias, alias object to be deleted
+     * @returns promise which will be resolve to operation object
+     * @memberof LogsAliasesService
+     */
+    deleteAlias (serviceName, alias) {
+        return this.AliasApiService.delete({ serviceName, aliasId: alias.aliasId }, alias)
+            .$promise
+            .then(operation => {
+                this._resetAllCache();
+                return this._handleSuccess(serviceName, operation, "logs_aliases_delete_success");
+            })
+            .catch(this.ServiceHelper.errorHandler("logs_aliases_delete_error"));
+    }
+
+    /**
+     * create new alias
+     *
+     * @param {any} serviceName
+     * @param {any} alias, alias object to be created
+     * @returns promise which will be resolve to operation object
+     * @memberof LogsAliasesService
+     */
+    createAlias (serviceName, alias) {
+        return this.AliasApiService.create({ serviceName }, alias)
+            .$promise
+            .then(operation => {
+                this._resetAllCache();
+                return this._handleSuccess(serviceName, operation, "logs_aliases_create_success");
+            })
+            .catch(this.ServiceHelper.errorHandler("logs_aliases_create_error"));
+    }
+
+    /**
+     * update alias
+     *
+     * @param {any} serviceName
+     * @param {any} alias, alias object to be updated
+     * @returns promise which will be resolve to operation object
+     * @memberof LogsAliasesService
+     */
+    updateAlias (serviceName, alias) {
+        return this.AliasApiService.update({ serviceName, aliasId: alias.aliasId }, alias)
+            .$promise
+            .then(operation => {
+                this._resetAllCache();
```

Comment:
Do you really need to reset all cache when you update an alias? why not only alias cache?

Reply:
yes, I need to reset 2 caches whenever the user modifies, adds or deletes an alias. LogsApiService is not required. I will remove it.
AliasAapiService cache needs to be cleared so that table shows latest data. AccountingAapiService needs to be cleared so that we get the latest information about max options and used options.